pkg/name/digest.go

```diff
@@ -45,14 +46,31 @@ func (d Digest) Identifier() string {
 	return d.DigestStr()
 }
 
+// TagStr returns the tag component of the Digest.
+func (d Digest) TagStr() string {
+	return d.tag
+}
+
 // DigestStr returns the digest component of the Digest.
 func (d Digest) DigestStr() string {
 	return d.digest
 }
 
 // Name returns the name from which the Digest was derived.
 func (d Digest) Name() string {
-	return d.Repository.Name() + digestDelim + d.DigestStr()
+	name := d.Repository.Name()
+	if d.tag != "" {
+		name += tagDelim + d.TagStr()
+	}
+	name += digestDelim + d.DigestStr()
+	return name
+}
+
+// Tag returns the tag component of the Digest (if any)
+func (d Digest) Tag() Tag {
+	name := d.Repository.Name() + tagDelim + d.TagStr()
+	tag, _ := NewTag(name, StrictValidation)
```

Comment:
Does this work properly when `TagStr()` returns `""`?

Reply:
No,  it returns a bogus result ending in `name@`. So it is tested before use (slightly ugly)

Reply:
Returning a `Tag{}` here would still require testing below, and returning `nil` would make it crash.
Adding an `error` only makes things worse, then it is no longer possible to use it as a one-liner.

Reply:
So basically what happens here, is that the new the `name` will fail the `StrictValidation`

And then the error will be ignored (`_`), and it (NewTag) will return the mentioned `Tag[}`

Reply:
This seems kind of error prone :/

Reply:
Also, this implementation doesn't preserve the original Digest options, see https://github.com/google/go-containerregistry/issues/563